### PR TITLE
Array delete

### PR DESCRIPTION
**Pull request by amosr**
! @jystic @tranma

### CODE REVIEW

File: icicle-compiler/src/Icicle/Sea/IO/Psv/Output.hs [hunk 1]
```diff
@@ -319,15 +321,15 @@ seaOfOutput isJSON struct structIndex outName@(OutputName name _) missing env ou
...
                    , outputMissing, body', ix, ts )
 
        -- Base
        _
         | (t  : ts) <- argTypes
         , (mx : _)  <- members
-        , mx'       <- transform mx
+        , mx'       <- transform t mx
         -> do d <- seaOfOutputBase' isJSON t mx'
               pure (Nothing, Nothing, d, structIndex + 1, ts)
```
Comment (tranma) [inline icicle-compiler/src/Icicle/Sea/IO/Psv/Output.hs]: checked out :chipmunk:

File: icicle-repl/test/cli/repl/t60-map-delete/script [hunk 2]
```diff
@@ -0,0 +1,10 @@
+:set +core-eval
+:set +c-eval
+
+feature injury ~> fold x = map_create : case map_lookup (Some location) x | None -> map_insert (Some location) (Some severity) x | Some _ -> map_delete (Some location) x end ~> x
+-- fold x = map_create :
+--  case map_lookup location x
+--  | None -> map_insert location severity x
+--  | Some _ -> map_delete location x
+--  end
```
Comment (tranma) [inline icicle-repl/test/cli/repl/t60-map-delete/script]: could we have this test case in bench as well? It's an interesting output type

**Comment by tranma** [comment]
☔️

File: icicle-repl/test/cli/repl/t60-map-delete/script (hunk 2 above)
Comment (amosr) [inline icicle-repl/test/cli/repl/t60-map-delete/script]: done

**Pull request merged**